pkg/server/server.go

```diff
@@ -491,8 +496,13 @@ func (s *ProxyServer) serveRecvFrontend(stream client.ProxyService_ProxyServer,
 			random := pkt.GetCloseDial().Random
 			klog.V(5).InfoS("Received DIAL_CLOSE", "serverID", s.serverID, "dialID", random)
 			// Currently not worrying about backend as we do not have an established connection,
-			s.PendingDial.Remove(random)
-			klog.V(5).InfoS("Removing pending dial request", "serverID", s.serverID, "dialID", random)
+			if pd := s.PendingDial.Remove(random); pd != nil {
+				klog.ErrorS(nil, "Dial cancelled (DIAL_CLS) by frontend",
+					"dialID", random,
+					"dialAddress", pd.dialAddress,
+					"dialDuration", time.Since(pd.start),
```

Comment:
Noticed most log lines in this function include serverID as a key, wondering if it's worthwhile keeping that here too